### PR TITLE
Add note on accessing last and best checkpoint

### PR DESCRIPTION
**Pull request by SkafteNicki**
## What does this PR do?

Fixes https://github.com/Lightning-AI/pytorch-lightning/issues/14740

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

----
📚 Documentation preview 📚: https://pytorch-lightning--21241.org.readthedocs.build/en/21241/

### CODE REVIEW

File: docs/source-pytorch/common/checkpointing_intermediate.rst [hunk 1]
```diff
@@ -21,7 +21,12 @@ For fine-grained control over checkpointing behavior, use the :class:`~lightning
         checkpoint_callback = ModelCheckpoint(dirpath="my/path/", save_top_k=2, monitor="val_loss")
         trainer = Trainer(callbacks=[checkpoint_callback])
         trainer.fit(model)
-        checkpoint_callback.best_model_path
```
Comment (deependujha) [inline docs/source-pytorch/common/checkpointing_intermediate.rst:24]: Any reason for removing this line?
It might be clearer to keep a code example showing both ways to access the checkpoint paths, e.g.:

```py
# Access directly from the callback
checkpoint_callback.best_model_path
checkpoint_callback.last_model_path

# Or via the trainer
trainer.checkpoint_callback.best_model_path
trainer.checkpoint_callback.last_model_path
```

This would make the docs more practical by demonstrating the options right in the example, alongside the note.

wdyt?

Comment (SkafteNicki) [reply docs/source-pytorch/common/checkpointing_intermediate.rst:24]: great suggestion! done

**Comment by deependujha** [comment]
The [CI failure](https://github.com/Lightning-AI/pytorch-lightning/pull/21241/checks?check_run_id=51198857043) looks like a timeout. The logs don’t explicitly say so, but the job ran for 57 minutes, and the timeout was recently changed from `75 to 55 minutes`. [check pr](https://github.com/Lightning-AI/pytorch-lightning/pull/21130/files)
